### ISSUE TITLE
onChange + onClick event not batched into same render commit

### ISSUE THREAD

**Issue by mischnic** [preactjs/preact#2745]
### Reproduction

https://codesandbox.io/s/preact-setstate-overwrite-input-moy82

### Steps to reproduce

(using `preact/compat`)

```js
function App() {
  console.log("rerender");
  let [checked, setChecked] = React.useState(false);

  let [, setState] = React.useState(false);
  return (
    <>
      <input
        type="checkbox"
        checked={checked}
        onChange={(e) => {
          console.log("onChange", e.target.checked);
          setChecked(e.target.checked);
        }}
        onClick={(e) => {
          console.log("onClick", e.target.checked);
          // ... or uncomment calls to fix
          setState(true);
          setState(false);
        }}
      />
      <br />
      Checked? {String(checked)}
    </>
  );
}
```

1. Try to toggle checkbox, doesn't work because this happens:
```
onClick  true
rerender 
onChange  false
```
So the click was overwritten by the `setState` call

2. With React:
```
onClick  true
onChange  true
rerender 
```
3. Without the `setState` calls, it obviously works

### Expected Behavior

The checkbox should toggle?

Not sure if this is something that can be fixed on the Preact side or if I need to add a workaround.
 
### Actual Behavior

Nothing happens when clicking the checkbox.

**Comment by marvinhagemeister** [comment]
We do batch all state updates which is why the `onClick` handler only causes one re-render. But having a closer look it seems like our window is smaller compared to React. A quick workaround would be to enlarge the time where updates are batched:

```js
import { options } from "preact";

options.debounceRendering = setTimeout;
```

**Comment by mischnic** [comment]
Thanks, though I'm not comfortable adding this line in a UI component library 😅

**Comment by iMakedonsky** [comment]
@marvinhagemeister  Thanks for this hack, it helps with following issue:

```
    "preact":  "^10.5.13",
    "@material-ui/core": "^4.11.3",
    "@material-ui/lab": "^4.0.0-alpha.57",
```

```
TypeError: Cannot read property 'focus' of null
handleClick [as clickfalse]
node_modules/@material-ui/lab/esm/useAutocomplete/useAutocomplete.js:919
  916 | 
  917 | 
  918 | var handleClick = function handleClick() {
> 919 |   inputRef.current.focus();
      | ^  920 | 
```

Which happens because somewhy onclick event gets processed in separate stack after the popup with element which was held in that `inputRef` was unmounted(there was onClick navigation on that element), so the `current` became null.

This works perfectly fine in React, probably because they process all events before next render cycle by default.

I guess this hack should become default behavior for `preact/compat` until there's a better solution.

**Comment by andrewiggins** [comment]
An alternative workaround for the original issue could be to call `setChecked()` in the `onClick()` handler. What I like about this "workaround" is that it more closely aligns with the DOM's behavior. In the `click` event, the DOM has changed the state of the checkbox, but in our code we don't sync our state to match the DOM's updated state. By calling `setChecked()` in the `onClick()` handler, our component's state is ensured to always be inline with the DOM's internal state.

Another thing to consider, I believe the `onClick()` and `onChange` events are largely duplicative of each other in the DOM. On my laptop and phone the `click` event is fired for mouse, keyboard, and touch interactions with the checkbox. So if the real code is specific to checkboxes, you could consider only using only one of `onClick` and `onChange`.

**Comment by JoviDeCroock** [comment]
Fixed in #3608
